--downloader.verify asserts for already completed files

I've missed an edge case as described in https://discord.com/channels/1133875232453693480/1356926710502916096/1446086688567722105.

```
panic: false

goroutine 1 [running]:
github.com/anacrolix/missinggo/v2/panicif.False(...)
    github.com/anacrolix/missinggo/v2@v2.10.0/panicif/panicif.go:73
github.com/erigontech/erigon/db/downloader.(*Downloader).addCompleteTorrentFromMetainfo(0x14000625200, {0x1400291faa0, 0x23}, 0x14000f98420)
    github.com/erigontech/erigon/db/downloader/downloader.go:1068 +0x1e0
github.com/erigontech/erigon/db/downloader.(*Downloader).addTorrentIfComplete(0x14000625200, {0x1400291faa0, 0x23})
    github.com/erigontech/erigon/db/downloader/downloader.go:1033 +0x1c0
github.com/erigontech/erigon/db/downloader.(*Downloader).AddTorrentsFromDisk.func2({0x1400291faa0, 0x2b}, {0x10350c8d8, 0x1400291dcc0}, {0x0, 0x0})
    github.com/erigontech/erigon/db/downloader/downloader.go:383 +0x1d8
io/fs.walkDir({0x1034ea3c0, 0x140020ddb20}, {0x1400291faa0, 0x2b}, {0x10350c8d8, 0x1400291dcc0}, 0x14001725ff0)
    io/fs/walk.go:73 +0x5c
io/fs.walkDir({0x1034ea3c0, 0x140020ddb20}, {0x140008eb190, 0x8}, {0x10350c8d8, 0x1400257e4c0}, 0x14001725ff0)
    io/fs/walk.go:95 +0x234
io/fs.walkDir({0x1034ea3c0, 0x140020ddb20}, {0x102ac1c18, 0x1}, {0x10350d020, 0x140020ddb30}, 0x14001725ff0)
    io/fs/walk.go:95 +0x234
io/fs.WalkDir({0x1034ea3c0, 0x140020ddb20}, {0x102ac1c18, 0x1}, 0x14001725ff0)
    io/fs/walk.go:122 +0x84
github.com/erigontech/erigon/db/downloader.(*Downloader).AddTorrentsFromDisk(0x14000625200, {0x10350ccc8, 0x1400181a500})
    github.com/erigontech/erigon/db/downloader/downloader.go:363 +0xc0
github.com/erigontech/erigon/node/eth.(*Ethereum).setUpSnapDownloader(0x14000530a88, {0x10350ccc8, 0x1400181a500}, 0x1400182e008, 0x1400041f800, 0x28?)
    github.com/erigontech/erigon/node/eth/backend.go:1328 +0x1ec
github.com/erigontech/erigon/node/eth.New({0x10350c540, 0x106194e40}, 0x14000522000, 0x14001d94008, {0x103528c50, 0x1400046b000}, 0x0)
    github.com/erigontech/erigon/node/eth/backend.go:418 +0x970
github.com/erigontech/erigon/cmd/erigon/node.New({0x10350c540, 0x106194e40}, 0x103528c50?, 0x14001d94008, {0x103528c50, 0x1400046b000}, 0x0)
    github.com/erigontech/erigon/cmd/erigon/node/node.go:159 +0x94
main.runErigon(0x14000f21b80)
    github.com/erigontech/erigon/cmd/erigon/main.go:95 +0x394
github.com/erigontech/erigon/cmd/utils/app.MakeApp.func1(0x14000f21b80)
    github.com/erigontech/erigon/cmd/utils/app/make_app.go:70 +0x108
github.com/urfave/cli/v2.(*Command).Run(0x14001d47080, 0x14000f21b80, {0x14000050240, 0x6, 0x6})
    github.com/urfave/cli/v2@v2.27.7/command.go:276 +0x580
github.com/urfave/cli/v2.(*App).RunContext(0x14000b3ee00, {0x10350c540, 0x106194e40}, {0x14000050240, 0x6, 0x6})
    github.com/urfave/cli/v2@v2.27.7/app.go:333 +0x4c8
github.com/urfave/cli/v2.(*App).Run(...)
    github.com/urfave/cli/v2@v2.27.7/app.go:307
main.main()
    github.com/erigontech/erigon/cmd/erigon/main.go:44 +0xb4
```

The workaround is to run without `--downloader.verify`, as you should only need that for Erigon 3.4+ if things are truly corrupted, rather than just the wrong `.torrent` file being there.

Only main should be affected.

@shohamc1

--- a/db/downloader/downloader.go
+++ b/db/downloader/downloader.go
@@ -1065,8 +1065,11 @@ func (d *Downloader) addCompleteTorrentFromMetainfo(
 			return
 		}
 	}
-	// TODO: Handle manual verification and ignore piece completion flags.
-	panicif.False(t.Complete().Bool())
+	if !d.cfg.VerifyTorrentData && !d.cfg.ManualDataVerification && !t.Complete().Bool() {
+		d.logger.Warn("expected completed snapshot but torrent client disagrees",
+			"snapshot name", name,
+			"infohash", t.InfoHash())
+	}
 	return
 }
 

